skxray/correlation.py

```diff
@@ -0,0 +1,319 @@
+    num_qs : int
+        number of region of interests(roi's)
+
+    pixel_list : ndarray
+        pixel indices for the required roi's
+
+    q_inds : ndarray
+        indices of the required roi's
+
+    img_stack : ndarray
+        intensity array of the images
+        dimensions are: (num of images, num_rows, num_cols)
+
+    Returns
+    -------
+    g2 : ndarray
+        matrix of one-time correlation
+        shape (num_levels, num_qs)
+
+    lag_steps : ndarray
+        delay or lag steps for the multiple tau analysis
+        shape num_levels
+
+    Notes
+    -----
+    In order to calculate correlations for delays, images must be
+    kept for up to the maximum delay. These are stored in the array
+    buf. This algorithm only keeps number of buffers and delays but
+    several levels of delays number of levels are kept in buf. Each
+    level has twice the delay times of the next lower one. To save
+    needless copying, of cyclic storage of images in buf is used.
+
+    References: text [1]_
+
+    .. [1] D. Lumma, L. B. Lurio, S. G. J. Mochrie and M. Sutton,
+        "Area detector based photon correlation in the regime of
+        short data batches: Data reduction for dynamic x-ray
+        scattering," Rev. Sci. Instrum., vol 70, p 3274-3289, 2000.
+
+    """
+    if num_bufs%2 != 0:
+        raise ValueError("number of channels or number of buffers in"
+                         " multiple-taus (must be even)")
+
+    # number of pixels in required roi's, dimensions are : [num_qs]X1
+    num_pixels = np.bincount(q_inds, minlength=(num_qs+1))
+    num_pixels = num_pixels[1:]
+
+    if np.any(num_pixels == 0):
+        raise ValueError("Number of pixels of the required roi's"
+                         " cannot be zero, "
+                         "num_pixels = {0}".format(num_pixels))
+
+    # matrix of auto-correlation function without normalizations
+    G = np.zeros(((num_levels + 1)*num_bufs/2, num_qs),
+                 dtype=np.float64)
+
+    # matrix of past intensity normalizations
+    IAP = np.zeros(((num_levels + 1)*num_bufs/2, num_qs),
+                   dtype=np.float64)
+
+    # matrix of future intensity normalizations
+    IAF = np.zeros(((num_levels + 1)*num_bufs/2, num_qs),
+                   dtype=np.float64)
+
+    # matrix of one-time correlation for required roi's
+    g2 = np.zeros((num_levels, num_qs), dtype=np.float64)
+
+    # correlation for delays, images must be keep for up to maximum
+    # delay in buf
+    buf = np.zeros((num_levels, num_bufs, np.sum(num_pixels)),
+                   dtype=np.float64)
+
+    # to track processing each level
+    cts = np.zeros(num_levels)
+
+    # to increment buffer
+    cur = np.ones((num_levels), dtype=np.int64)
+
+    # to track how many images processed in each level
+    num = np.array(np.zeros(num_levels), dtype=np.int64)
+
+    # starting time for the process
+    t1 = time.time()
+
+    for n in range(0, len(img_stack)):  # changed the number of frames
```

Comment:
Do this as 

```python
for n, img in enumerate(img_stack):
```
Because you don't need to know how many images there are.  This changes means we can pass generators in to this function.